test/t/index/test_dump_and_load_index.cpp

```diff
@@ -0,0 +1,45 @@
+#include "catch.hpp"
+
+#include <osmium/index/detail/tmpfile.hpp>
+#include <osmium/index/map/dense_file_array.hpp>
+#include <osmium/index/map/dense_mmap_array.hpp>
+#include <osmium/osm/location.hpp>
+#include <osmium/osm/types.hpp>
+
+using dense_mmap_array_t = osmium::index::map::DenseMmapArray<osmium::unsigned_object_id_type, osmium::Location>;
+using dense_file_array_t = osmium::index::map::DenseFileArray<osmium::unsigned_object_id_type, osmium::Location>;
+
+TEST_CASE("Dump DenseMmapArray, load as DenseFileArray") {
+    const int fd = osmium::detail::create_tmp_file();
+    REQUIRE(osmium::file_size(fd) == 0);
+    constexpr const size_t S = sizeof(dense_file_array_t::element_type);
+
+    const osmium::unsigned_object_id_type id1 = 12;
+    const osmium::unsigned_object_id_type id2 = 3;
+    const osmium::unsigned_object_id_type id3 = 7;
+    const osmium::Location loc1{1.2, 4.5};
+    const osmium::Location loc2{3.5, -7.2};
+    const osmium::Location loc3{-12.7, 14.5};
+
+    dense_mmap_array_t index;
+    index.set(id1, loc1);
+    index.set(id2, loc2);
+    index.set(id3, loc3);
+
+    // We don't sort the index because sorting is not required for DenseMmapArray.
+    // dump to file
+    REQUIRE_NOTHROW(index.dump_as_array(fd));
```

Comment:
Not sure what the `REQUIRE_NOTHROW` helps here. If it throws the program will crash, so the test will fail.

Reply:
removed